jsre/jsre.go

```diff
@@ -70,6 +73,28 @@ func New(assetPath string) *JSRE {
 	return re
 }
 
+// randomSource returns a pseudo random float64 in the range [0.0-1.0).
+func randomSource() float64 {
+	var src rand.Source
+	var once sync.Once
+
+	once.Do(func() {
+		bytes := make([]byte, 8)
+		seed := time.Now().UnixNano()
+		if _, err := crand.Read(bytes); err == nil {
+			seed = int64(binary.LittleEndian.Uint64(bytes))
+		}
+		src = rand.NewSource(seed)
+	})
+
+again:
+	f := float64(src.Int63()) / (1 << 63)
+	if f == 1 {
+		goto again // resample; this branch is taken O(never)
+	}
```

Comment:
I don't think this si correct here, buy maybe I misunderstand.

`src.Int63()` should return an 63 bit random integer. That is, between `0` and `2^64 - 1`. If you divide by `1 << 64` it should produce the intended results `[0, 1)`. Whereas dividing by `1 << 63` actually produces quite a lot of values over 1.